datalad/metadata/metadata.py

```diff
@@ -561,9 +566,36 @@ def _get_metadata(ds, types, global_meta=None, content_meta=None, paths=None):
 
         unique_cm = {}
         extractor_unique_exclude = getattr(extractor_cls, "_unique_exclude", set())
+        # TODO: ATM neuroimaging extractors all provide their own internal
+        #  log_progress but if they are all generators, we could provide generic
+        #  handling of the progress here.  Note also that log message is actually
+        #  seems to be ignored and not used, only the label ;-)
```

Comment:
The message are shown when `is_interactive()` is false.

```python
import logging
from datalad.log import log_progress

lgr = logging.getLogger("datalad.test")
log_progress(lgr.debug, "id", "message, start", label="label", unit=" unit")
log_progress(lgr.debug, "id", "message, update", update=1, increment=True)
log_progress(lgr.debug, "id", "message, done")
```

```
% DATALAD_LOG_LEVEL=debug python scratch.py | cat
[DEBUG] message, start
[DEBUG] message, update
[DEBUG] message, done
```

Reply:
I was blind, thanks!